lib/web/inventory.go

```diff
@@ -0,0 +1,161 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package web
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+
+	inventoryv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/inventory/v1"
+	"github.com/gravitational/teleport/lib/defaults"
+	"github.com/gravitational/teleport/lib/reversetunnelclient"
+	"github.com/gravitational/teleport/lib/web/ui"
+)
+
+// listUnifiedInstancesResponse is the response for listing unified instances
+type listUnifiedInstancesResponse struct {
+	// Instances is the list of unified instances (both instances and bot instances)
+	Instances []ui.UnifiedInstance `json:"instances"`
+	// StartKey is the next page token
+	StartKey string `json:"startKey"`
+}
+
+// clusterUnifiedInstancesGet returns a paginated list of unified instances
+func (h *Handler) clusterUnifiedInstancesGet(w http.ResponseWriter, r *http.Request, p httprouter.Params, sctx *SessionContext, cluster reversetunnelclient.Cluster) (any, error) {
+	clt, err := sctx.GetUserClient(r.Context(), cluster)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	values := r.URL.Query()
+
+	limit, err := QueryLimitAsInt32(values, "limit", defaults.MaxIterationLimit)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	startKey := values.Get("startKey")
+
+	sortParam := values.Get("sort")
+	var sort inventoryv1.UnifiedInstanceSort
+	var order inventoryv1.SortOrder
+
+	// Default values
+	sort = inventoryv1.UnifiedInstanceSort_UNIFIED_INSTANCE_SORT_NAME
+	order = inventoryv1.SortOrder_SORT_ORDER_ASCENDING
+
+	if sortParam != "" {
+		parts := strings.SplitN(sortParam, ":", 2)
+		fieldName := strings.ToLower(parts[0])
+
+		switch fieldName {
+		case "name", "hostname":
+			sort = inventoryv1.UnifiedInstanceSort_UNIFIED_INSTANCE_SORT_NAME
+		case "type":
+			sort = inventoryv1.UnifiedInstanceSort_UNIFIED_INSTANCE_SORT_TYPE
+		case "version":
+			sort = inventoryv1.UnifiedInstanceSort_UNIFIED_INSTANCE_SORT_VERSION
+		}
+
+		if len(parts) == 2 {
+			direction := strings.ToLower(parts[1])
+			if direction == "desc" || direction == "descending" {
+				order = inventoryv1.SortOrder_SORT_ORDER_DESCENDING
+			}
+		}
+	}
+
+	filter := &inventoryv1.ListUnifiedInstancesFilter{}
+
+	if search := values.Get("search"); search != "" {
+		filter.Search = search
+	}
+
+	if query := values.Get("query"); query != "" {
+		filter.SearchKeywords = query
+	}
```

Comment:
This would be equivalent, right?
```suggestion
	filter := &inventoryv1.ListUnifiedInstancesFilter{
		Search:         values.Get("search"),
		SearchKeywords: values.Get("query"),
	}
```